Added huggingface model loader/importer to export.py

atamurad opened the pull request:
Summary:
 * Added `load_hf_model()` function to export.py
 * Correctly set `shared_classifier` flag and write classifier weights if exists, in all export functions
 * Added `hidden_dim` to `ModelArgs`
 * Added `--hf` CLI argument to load HF model

Tests:
 * Legacy export with Llama-2-7b-chat-hf model - OK
 * model.py inference with generate() - OK

Fixes:
 * https://github.com/karpathy/llama2.c/issues/314
 * WQ / WK permutation bug reported in comments: https://github.com/karpathy/llama2.c/pull/286#issuecomment-1679066644

karpathy commented:
@atamurad this looks nice to me. is it ready to merge? the only thing i'd maybe add is we should take out transformers from requirements? it feels like a really heavy dependency for a niche import feature that maybe isn't going to be all that widely used for the core repo? Sometimes I just put a note next to the import giving the pip install command or something, so if someone gets an ImportError they see right away how to fix it.
?

atamurad commented:
@karpathy yeah wasn't sure about including it in requirements.txt, just removed it and added error message. is phrasing ok? 

otherwise I think it's good to merge.

karpathy commented:
great! I'll aim to merge tonight.

karpathy commented:
Cool PR thank you!
